app/views/admin/products_v3/_variant_row.html.haml

```diff
@@ -56,6 +57,7 @@
       include_blank: t('.none_tax_category'),
       aria_label: t('.tax_category_field_name'),
       placeholder_value: t('.search_for_tax_categories')))
+  = error_message_on variant, :tax_category_id
```

Comment:
I just had a play with this and it seemed to work to go back to validating tax_category and just changing the error message here, without changing the input above:
```suggestion
  = error_message_on variant, :tax_category
```

If I require a tax category but don't have a default then this will display an error.

The grey zone is to require a tax category, set a default and then select "None". Then it will look like it doesn't have a tax category but the default will be applied during checkout. That's another problem though.

Reply:
Nice work, I didn't think of that. 

It won't work for the above select (ie `<select name="...variant[tax_category_id]">`). That means we won't get the `field_with_errors` wrapper class to add a red border on the field.
But that's an acceptable tradeoff 👍

Reply:
Note to self: also add spec for this error.